source/distributed/tria_base.cc

```diff
@@ -342,6 +351,196 @@ namespace parallel
 
 
 
+  template <int dim, int spacedim>
+  void
+  TriangulationBase<dim, spacedim>::reset_global_cell_indices()
+  {
+#ifndef DEAL_II_WITH_MPI
+    Assert(false, ExcNeedsMPI());
+#else
+
+    // currently only implemented for distributed triangulations
+    if (dynamic_cast<const parallel::DistributedTriangulationBase<dim, spacedim>
+                       *>(this) == nullptr)
+      return;
+
+    // 1) determine number of active locally-owned cells
+    const types::global_cell_index n_locally_owned_cells =
+      this->n_locally_owned_active_cells();
+
+    // 2) determine the offset of each process
+    types::global_cell_index cell_index = 0;
+
+    MPI_Exscan(&n_locally_owned_cells,
+               &cell_index,
+               1,
+               Utilities::MPI::internal::mpi_type_id(&n_locally_owned_cells),
+               MPI_SUM,
+               this->mpi_communicator);
+
+    // 3) give global indices to locally-owned cells and mark all other cells as
+    //    invalid
+    for (const auto &cell : this->active_cell_iterators())
+      if (cell->is_locally_owned())
+        cell->set_active_cell_index(cell_index++);
+      else
+        cell->set_active_cell_index(numbers::invalid_dof_index);
```

Comment:
don't we have an `invalid_cell_index`? This has the wrong type...

Reply:
also, don't you want to set the *global* indices here and below? This looks wrong.